logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -202,38 +205,41 @@ class ConversationDataSource(
 
     private suspend fun getConversationDetailsFlow(conversation: Conversation): Flow<ConversationDetails> = when (conversation.type) {
         Conversation.Type.SELF -> flowOf(ConversationDetails.Self(conversation))
-        Conversation.Type.GROUP -> flowOf(
-            ConversationDetails.Group(
-                conversation, LegalHoldStatus.DISABLED // TODO(user-metadata): get actual legal hold status
-            )
-        )
-        // TODO(connection-requests): Handle requests instead of filtering them out
+        // TODO(user-metadata): get actual legal hold status
+        Conversation.Type.GROUP -> flowOf(ConversationDetails.Group(conversation, LegalHoldStatus.DISABLED))
         Conversation.Type.CONNECTION_PENDING, Conversation.Type.ONE_ON_ONE -> {
             val selfUser = userRepository.observeSelfUser().first()
+            when (val conversationMembers = getConversationMembers(conversation.id)) {
```

Comment:
why we go here with this approach to create a variable and pass it to the function 
isn't keep the fold here and pass the actual value not the Either to the success and error functions  better?

Reply:
I found it to be more clear in terms of readability and helped to debug better. But doesn't make any difference in terms of the bug fix. 

I'll try to find a better way to write it, if I'm not able, I'll just rollback these changes =D